### ISSUE TITLE
Workflow for WordPress.com sites with Latin1 database encoding

### ISSUE THREAD

**Issue by daledupreez** [blogvault/wpcom-migration#60]
For longstanding legacy reasons, WordPress.com still uses Latin1 encoding for sites on the platform. We try to initialise all sites expecting migrations with UTF-8 databases, but there are a number of flows and pathways whereby we can't do that.

Is it possible for us to set up a workflow or check for the WordPress.com site's database charset prior to us starting the migration process? From our side, we'd love to be able to automate getting the site switched across to UTF-8, but an early first step may be to create a support issue on WordPress.com to help with this.

I am not sure what options we have for that from the Blog Vault side, so this issue is a way for us to discuss that.

**Comment by bv-atulya** [comment]
Hi @daledupreez 

I have a couple of questions regarding the requirements:

1. When referring to UTF-8 databases, are you specifically indicating that the database should use utf8mb4 encoding? Currently, we use the query `SHOW CHARACTER SET LIKE 'utf8mb4'` to verify UTF-8 usage in the database. Would this approach be suitable in this context as well?

2. For Version 1, would you prefer that we automatically create a support ticket if an issue arises, or should we halt the migration with a clear, user-friendly error message instead?

**Comment by bv-shivamsharma** [comment]
Hi @daledupreez Could you please help @bv-atulya in answering his questions?

**Comment by sixhours** [comment]
Hi @bv-shivamsharma! Dale is on sabbatical for the next three months, but I will look at getting answers to these questions and get back to you before the end of the week. Thank you!

**Comment by sixhours** [comment]
> When referring to UTF-8 databases, are you specifically indicating that the database should use utf8mb4 encoding? Currently, we use the query SHOW CHARACTER SET LIKE 'utf8mb4' to verify UTF-8 usage in the database. Would this approach be suitable in this context as well?

Yes, that's the character set we're looking for, but that query will only show if utf8mb4 is _available_ on the server, not necessarily the schema being used. You'd probably need to access the `INFORMATION_SCHEMA` table for that.

> For Version 1, would you prefer that we automatically create a support ticket if an issue arises, or should we halt the migration with a clear, user-friendly error message instead?

I don't think you'd need to halt the migration or show an error. If a site has a UTF-8 encoded database, you can send it as a field in the payload response when a migration has `STARTED` status and we'll handle creating the support ticket on our end.

Let me know if this helps!

**Comment by bv-shivamsharma** [comment]
@sixhours Thank you for the information, we'll let you know once we are ready with the code.

**Comment by markbiek** [comment]
Hi @bv-shivamsharma. It doesn't seem like including the database schema in the `STARTED` payload was implemented.

Do you have any additional context you could share?

**Comment by markbiek** [comment]
I just wanted to note that this has been implemented and deployed.
